src/Workspaces/CSharp/Portable/Editing/CSharpImportAdder.cs

```diff
@@ -70,12 +70,15 @@ protected override SyntaxNode InsertNamespaceImport(SyntaxNode root, SyntaxGener
                     : UsingsAndExternAliasesDirectiveComparer.NormalInstance;
 
             // find insertion point
+            int existingIndex = 0;
             foreach (var existingImport in gen.GetNamespaceImports(root))
             {
                 if (comparer.Compare(import, existingImport) < 0)
                 {
-                    return gen.InsertNodesBefore(root, existingImport, new[] { import });
```

Comment:
It turns out that the existing downstream call to PreserveTrivia is sufficient for this to work. So I've reverted this change and added a test.